PURL-TYPES.rst

```diff
@@ -253,6 +253,23 @@ github
       pkg:github/package-url/purl-spec@244fd47e07d1004
       pkg:github/package-url/purl-spec@244fd47e07d1004#everybody/loves/dogs
 
+githubactions
+------
+``githubactions`` for GitHub Actions:
+
+- The ``namespace`` is the user or organization. It is not case sensitive and
+  must be lowercased.
+- The ``name`` is the repository name. It is not case sensitive and must be
+  lowercased.
+- The ``subpath`` is used to point to a secondary action within a repository. It is not case sensitive and must be lowercased.
+- The ``version`` is a commit or release tag. 
+- Examples::
+
+      pkg:githubactions/package-url/purl-spec@244fd47e07d1004
+      pkg:githubactions/package-url/purl-spec@v1.2
+      pkg:githubactions/github/codeql-action/analyze@v2
```

Comment:
What is the relationship between `pkg:githubactions/github/codeql-action/analyze@v2` and the Marketplace where actions are found? For example, this action is not listed in the marketplace, but https://github.com/marketplace/actions/codeql-bundle is. I think this needs to be clarified as many people will associate the `githubactions` purl type with the actions that are in the Marketplace, but that's not true in this case.

Reply:
The Marketplace listings are basically advertisements and disconnected from the actual software distribution. You can reference an action in any repository regardless of whether there is a Marketplace listing. The only requirement is that there's an action.yml file in the repository and that the version matches a SHA, branch, or release tag. You'll see some of the scenarios around how things get referenced here: https://docs.github.com/en/actions/learn-github-actions/finding-and-customizing-actions#adding-an-action-to-your-workflow